lib/IAMClient.js

```diff
@@ -284,40 +284,76 @@ class VaultClient {
     /**
      * Update Quota of an account
      *
-     * @param {string} accountName - account name
      * @param {number} quota - maximum quota for the account
+     * @param {object} options - additional creation params
+     * @param {string} [options.accountName] - account name
      * @param {VaultClient~requestCallback} callback - callback
      * @returns {undefined}
      */
-    updateAccountQuota(accountName, quota, callback) {
-        const conv = Number.isNaN(Number.parseInt(quota, 10));
-        assert(typeof accountName === 'string' && accountName !== '',
-            'accountName is required');
-        assert(typeof quota === 'number' && !conv
-            && quota >= 0, 'Quota must be a non-negative number');
-        this.request('POST', '/', true, callback, {
+    updateAccountQuota(quota, options, callback) {
+        const { accountName } = options || {};
+        const data = {
             Action: 'UpdateAccountQuota',
             Version: '2010-05-08',
-            AccountName: accountName,
             quotaMax: quota,
-        });
+        };
+
+        const quotaIsValid = typeof quota === 'number' && !Number.isNaN(Number.parseInt(quota, 10)) && quota > 0;
+        assert(quotaIsValid, 'Quota must be a strictly positive number');
+        if (accountName) {
+            assert(typeof accountName === 'string', 'the account name, if set, should be a string');
+            data.AccountName = accountName;
+        }
+        this.request('POST', '/', true, callback, data);
     }
 
     /**
      * Delete Quota of an account
      *
-     * @param {string} accountName - account name
+     * @param {object} options - additional creation params
+     * @param {string} [options.accountName] - account name
+     *  set to true, it helps sending invalid requests, to ensure
+     *  input validation.
      * @param {VaultClient~requestCallback} callback - callback
      * @returns {undefined}
      */
-    deleteAccountQuota(accountName, callback) {
-        assert(typeof accountName === 'string' && accountName !== '',
-            'accountName is required');
-        this.request('POST', '/', true, callback, {
+    deleteAccountQuota(options, callback) {
+        const data = {
             Action: 'DeleteAccountQuota',
             Version: '2010-05-08',
-            AccountName: accountName,
-        });
+        };
+        if (options !== undefined && Object.keys(options).length !== 0) {
```

Comment:
this means that if no options are passed, the call with succeed : is `accountName` really optional? does it make sense to call `deleteAccountQuota` without specifying the account?

Reply:
this has been removed , now we do check directly if the accountName is passed to the function see here : https://github.com/scality/vaultclient/pull/442/files#diff-86a8509a2ad22b19f93123bda4efc46cfa71b4c2bbfded93e5f6e1a25e69f180R330